lib/libzpool/Makefile.am

```diff
@@ -17,6 +17,7 @@ AM_CFLAGS += $(NO_UNUSED_BUT_SET_VARIABLE)
 # Includes kernel code generate warnings for large stack frames
 AM_CFLAGS += $(FRAME_LARGER_THAN)
 
+# frame-larger-than=20480 is needed here due to requirements of the stock zstd
 AM_CFLAGS += -DLIB_ZPOOL_BUILD -Wframe-larger-than=20480
```

Comment:
Is this really needed both in the libzpool makefile as the module makefile?

Reply:
lemme try to build without

Reply:
we get warnings here without it. setting it at this place changes the value for whole libzpool, which is not what we want... we only want to set this for the zstd stuff but I don't know if that's possible :S

Reply:
frame-larger-then must be 2048. this is a typo

Reply:
otherwise you get compiler warnings. this is related to the zstd library which is linked to userspace code and compiled from libzpool in userspace mode. however. in kernel mode you need that too. just in zstd.c this time only

Reply:
@brainslayer You mean its a typo here, or in both places?

Reply:
but you can check locally which value is best. for x86 its 1024, for x64 its a little bit higher (something around 1200). but for arm64 its close to 2048. i have choosen 2048 as safe value to suppress compiler warnings

Reply:
mmh wait. i dont have this line in my local tree. i believe this is only required for kernel mode compile since the kernel has a lower limit by default. you can take this out from libzpool. its also not in my so called "reference tree" :-)

Reply:
@brainslayer, so the 20480 is only for the zstd module makefile?

Reply:
@c0d3z3r0 So this can be removed here and left on the module it seems...

Reply:
I get a ton of warning, if I remove it as commented above

Reply:
@c0d3z3r0 Even if you just leave this one at default?

Reply:
I got warnings about 16k in both cases but I will do a new test with a complete clean build

Reply:
@c0d3z3r0 To prevent endless discussions with reviewers, lets make it like this:
`# Due to stock zstd using some 16k frames frame-larger-than=20480 is required`

This also gives the literal reason WHY it is required for ZSTD stock ;)

Reply:
sounds good, I just want to verify that once again